src/poetry/factory.py

```diff
@@ -74,10 +74,18 @@ def create_poetry(
         )
 
         # Configuring sources
-        self.configure_sources(
-            poetry, poetry.local_config.get("source", []), config, io
+        config.merge(
+            {
+                "sources": {
+                    source["name"]: source
+                    for source in poetry.local_config.get("source", [])
+                }
+            }
         )
 
+        # Configuring sources
```

Comment:
These comments are duplicates, and the `merge()` call is a bit overnested. I think we should break the generator expression out into a separate line (e.g. L98 in `add.py`).